src/renderers/WebGLRenderer.js

```diff
@@ -701,13 +708,14 @@ function WebGLRenderer( parameters ) {
 		state.setMaterial( material, frontFaceCW );
 
 		var program = setProgram( camera, fog, material, object );
-		var geometryProgram = geometry.id + '_' + program.id + '_' + ( material.wireframe === true );
 
 		var updateBuffers = false;
 
-		if ( geometryProgram !== _currentGeometryProgram ) {
+		if ( geometry.id !== _currentGeometryProgram.geometryID || program.id !== _currentGeometryProgram.programID || material.wireframe !== _currentGeometryProgram.wireframe ) {
 
-			_currentGeometryProgram = geometryProgram;
+			_currentGeometryProgram.geometryID = geometry.id;
+			_currentGeometryProgram.programID = program.id;
+			_currentGeometryProgram.wireframe = material.wireframe;
```

Comment:
Some `Material`s don't have `.wireframe` so here can be `undefined`. This could be better?

    _currentGeometryProgram.wireframe = material.wireframe === true;

Reply:
Yes, makes sense! Should this be address in this PR or another PR? I already removed the branch.

Reply:
I think another PR is fine.

Reply:
I can do the change locally 👌

Reply:
Also, instead of this..


```js
_currentGeometryProgram.geometry = geometry.id;
_currentGeometryProgram.program = program.id;
```

couldn't it just be...

```js
_currentGeometryProgram.geometry = geometry;
_currentGeometryProgram.program = program;
```

Reply:
Note: If you wanna do, you should clear `_currentGeometryProgram` in the end of `.render()`. Unless doing that, `_currentGeometryProgram` keeps the reference to `geometry` and `program` until calling next `.render()`. That means, even if you no longer need the `geometry/program` the resources aren't released without calling next `.render()`.